src/codeu/chat/client/commandline/Chat.java

```diff
@@ -481,4 +766,144 @@ public void invoke(List<String> args) {
     // so that it can be used.
     return panel;
   }
+
+  private void updateConversationsMap(Uuid user, Uuid convo){
+    Set<Uuid> conversations = (updatedConversationsMap.get(user) == null) ? new HashSet<>() : updatedConversationsMap.get(user);
+
+    conversations.add(convo);
+    updatedConversationsMap.put(user, conversations);
+
+  }
+
+  private void removeUpdatedConversation(Uuid user, Uuid convo) {
+    if(updatedConversationsMap.containsKey(user) && updatedConversationsMap.get(user) != null){
+      Set<Uuid> conversations = updatedConversationsMap.get(user);
+      conversations.remove(convo);
+      updatedConversationsMap.put(user,conversations);
+    }
+
+  }
+
+  // Increments the message count of a specified conversation of interest
+  // of a specific user.
+  private void incrementMessageCount(Uuid convoID){
+
+    for(UserContext user : rootPanelContext.allUsers()){
+      HashMap<Uuid, Integer> convoCount = (convoMessageCountsMap.get(user.user.id) == null) ? new HashMap<>() : convoMessageCountsMap.get(user.user.id);
+      Integer count = (convoCount.get(convoID) == null) ? 0 : convoCount.get(convoID);
+
+      convoCount.put(convoID, count + 1);
+      convoMessageCountsMap.put(user.user.id, convoCount);
+
+    }
+  }
+
+  // Getter method for conversation message counts in conversations of interest.
+  private Integer getMessageCount(Uuid userID, Uuid convoID){
+    return (convoMessageCountsMap.get(userID) == null) ? 0 : convoMessageCountsMap.get(userID).get(convoID);
+  }
+
+  //methods below are helper methods to get a user or conversation from name or Uuid
+
+  // Find the first user with the given name and return a user context
+  // for that user. If no user is found, the function will return null.
+  private UserContext findUser(String name) {
+    for (final UserContext user : rootPanelContext.allUsers()) {
```

Comment:
Same thing with `.allUsers()` here, if it can return a `Set`, you can speed this method up.